### PR TITLE
add Foreman plugin monitoring

### PR DESCRIPTION
**Pull request by dgoetz**
Please build for

* [x] Nightly
* [x] 1.12
* [ ] 1.11
* [ ] 1.10

### CODE REVIEW

**Comment by domcleal** [comment]
Please ignore the test result for now, it's been muddled with another PR. It is building OK, I'll re-run it later.

The plugin ships app/overrides/ with some Deface overrides, but the gem needs to depend on `deface` for this to work. At the moment it fails at runtime as deface is missing. You'll need to fix this in the gem and release an update.

**Comment by dgoetz** [comment]
I did an new release and pump also the version of the spec. Not sure if you have to run the tests manually now.

**Comment by domcleal** [comment]
Thanks, they'll run eventually once the queue on the Jenkins server has reduced.

File: rubygem-foreman_monitoring/rubygem-foreman_monitoring.spec [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+%{gem_instdir}/test
+
+%posttrans
+%{foreman_db_migrate}
+%{foreman_db_seed}
+%{foreman_restart}
+exit 0
+
+%changelog
+* Fri Aug 19 2016 Dirk Goetz <dirk.goetz> - 0.0.1-1
```
Comment (domcleal) [inline rubygem-foreman_monitoring/rubygem-foreman_monitoring.spec]: 0.0.2

**Comment by domcleal** [comment]
Can you add the deface dependency to both Requires and BuildRequires please? It should mirror the gemspec, i.e. `rubygem(deface) < 2.0`

**Comment by domcleal** [comment]
There remains a problem on app startup that Deface isn't found, it looks like engine.rb is missing a `require 'deface'` to load the library. I don't think the dependency is automatically loaded through bundler_ext.

<pre>
[ 2016-08-23 03:33:58.7769 1165/7f85afb80700 Pool2/Implementation.cpp:287 ]: Could not spawn process for application /usr/share/foreman: An error occured while starting up the preloader.
  Error ID: 6c20936f
  Error details saved to: /tmp/passenger-error-Jl0Qh3.html
  Message from application: uninitialized constant Deface (NameError)
  /opt/theforeman/tfm/root/usr/share/gems/gems/foreman_monitoring-0.0.2/app/overrides/add_host_monitoring_result_tab.rb:1:in `<top (required)>'
  /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
  /opt/rh/rh-ruby22/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
</pre>

**Comment by dgoetz** [comment]
I think it messed up again with the test results, because on koji it build fine after I added deface and bump the version.

**Comment by domcleal** [comment]
Thanks for checking, I've run the test manually and it's building fine. Merged and building now, thanks @dgoetz.

**Pull request merged**